### PR TITLE
Update links with query params

### PR DESCRIPTION
**Pull request by gkillough**

### CODE REVIEW

**Comment by ghost** [comment]
## Black Duck Security Report

Merging #78 into master will not change security risk.

### Added Components

Clean:       3

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/blackducksoftware/blackduck-jira/pull-requests/78)

**Review by jamesrichard91 — APPROVED** [review]

File: src/main/java/com/blackducksoftware/integration/jira/task/issue/handler/DataFormatHelper.java [hunk 1]
```diff
@@ -102,17 +105,21 @@ public String getIssueDescription(final IssueCategory issueCategory, final Strin
         return issueDescription.toString();
     }
 
-    public String createVulnerableComponentsLink(final String componentName, final String baseVulnerableComponentsLink) {
-        if (StringUtils.isNotBlank(componentName)) {
-            final String encodedComponentName;
+    public Optional<String> createUrlWithComponentNameQuery(final String url, final String componentName) {
+        if (StringUtils.isNotBlank(componentName) && StringUtils.isNotBlank(componentName)) {
```
Comment (jamesrichard91) [inline src/main/java/com/blackducksoftware/integration/jira/task/issue/handler/DataFormatHelper.java]: Why is this if checking the same thing twice?

File: src/main/java/com/blackducksoftware/integration/jira/task/issue/handler/DataFormatHelper.java [hunk 2]
```diff
@@ -102,17 +105,21 @@ public String getIssueDescription(final IssueCategory issueCategory, final Strin
...
-                encodedComponentName = URLEncoder.encode(componentName, "UTF-8");
-                return String.format("%s?q=componentName:%s", baseVulnerableComponentsLink, encodedComponentName);
-            } catch (UnsupportedEncodingException e) {
-                logger.debug("Error encoding vulnerable components link: " + e.getMessage());
+                // Black Duck does not encode query parameters in the traditional way (UTF-8), so we must do a little extra work to ensure the encoding will match.
+                final String unencodedUrl = String.format("%s?q=componentName:%s", url, componentName);
+                URL encodedUrl = new URL(unencodedUrl);
+                URI uri = new URI(encodedUrl.getProtocol(), encodedUrl.getUserInfo(), IDN.toASCII(encodedUrl.getHost()), encodedUrl.getPort(), encodedUrl.getPath(), encodedUrl.getQuery(), encodedUrl.getRef());
+
+                final String asciiString = uri.toASCIIString();
```
Comment (jamesrichard91) [inline src/main/java/com/blackducksoftware/integration/jira/task/issue/handler/DataFormatHelper.java:117]: Do you want to include the stack overflow link in a comment to explain this method of encoding?

File: src/main/java/com/blackducksoftware/integration/jira/task/issue/handler/DataFormatHelper.java (hunk 1 above)
Comment (gkillough) [reply src/main/java/com/blackducksoftware/integration/jira/task/issue/handler/DataFormatHelper.java]: Oops, I refactored the variable order, but somehow screwed this up.

**Review by bamandel — APPROVED** [review]

File: src/main/java/com/blackducksoftware/integration/jira/task/issue/handler/DataFormatHelper.java (hunk 2 above)
Comment (psantos1113) [reply src/main/java/com/blackducksoftware/integration/jira/task/issue/handler/DataFormatHelper.java:117]: I've unfortunately had to do the same thing before to ensure proper encoding as well in the past.

**Review by psantos1113 — APPROVED** [review]

**Pull request merged**